packages/gestalt/src/Checkbox/InternalCheckbox.tsx

```diff
@@ -180,10 +184,41 @@ const InternalCheckboxWithForwardRef = forwardRef<HTMLInputElement, Props>(funct
     [styles.readOnly]: readOnly,
   });
 
+  const iconWrapperStyles = classnames(styles.vrCheckIconWrapper, {
+    [styles.vrCheckIconEnterTransition]: !unchecked,
+    [styles.vrCheckIconExitTransition]: unchecked,
+  });
+
   return (
     <Box>
       <Box alignItems="start" display="flex" justifyContent="start" marginEnd={-1} marginStart={-1}>
         <Box paddingX={1} position="relative">
+          <div ref={tapScaleAnimation.elementRef} className={divStyles} style={style}>
+            {isInVRExperiment ? (
+              <div style={{ width: vrIconSizes[size] }}>
+                <div
+                  className={iconWrapperStyles}
+                  style={{ width: checked || indeterminate ? vrIconSizes[size] : 0 }}
+                >
+                  <Icon
+                    accessibilityLabel=""
+                    color={isInVRExperiment ? vrIconColor : 'inverse'}
+                    icon={indeterminate ? 'dash' : 'check'}
+                    size={isInVRExperiment ? vrIconSizes[size] : iconSizes[size]}
+                  />
+                </div>
+              </div>
+            ) : (
+              (checked || indeterminate) && (
+                <Icon
+                  accessibilityLabel=""
```

Comment:
isInVRExperiment is redundant here, right? @diyorbek

Reply:
oh yes, right